sklearn/linear_model/quantile.py

```diff
@@ -0,0 +1,314 @@
+
+    xtol : float, default 1e-6
+        Global optimization will stop when ``|w_{t-1} - w_t|`` < ``xtol``
+        where w_t is result of t'th approximated optimization.
+
+    gamma_decrease : float, default 0.1
+        The factor by which ``gamma`` is multiplied at each iteration.
+
+    n_gamma_decreases : int, default 10
+        Maximal number of iterations of approximation of the cost function.
+        At each iteration, ``gamma`` is multiplied by a factor
+        of ``gamma_decrease``
+
+    Attributes
+    ----------
+    coef_ : array, shape (n_features,)
+        Features got by optimizing the Huber loss.
+
+    intercept_ : float
+        Bias.
+
+    n_iter_ : int
+        Number of iterations that scipy.optimize.mimimize has run for.
+
+    References
+    ----------
+    .. [1] Koenker, R., & Bassett Jr, G. (1978). Regression quantiles.
+            Econometrica: journal of the Econometric Society, 33-50.
+
+    .. [2] Chen, C., & Wei, Y. (2005).
+           Computational issues for quantile regression.
+           Sankhya: The Indian Journal of Statistics, 399-417.
+    """
+
+    def __init__(self, quantile=0.5,
+                 max_iter=10000, alpha=0.0001, l1_ratio=0.0,
+                 warm_start=False, fit_intercept=True,
+                 normalize=False, copy_X=True,
+                 gamma=1e-2, gtol=1e-4, xtol=1e-6,
+                 gamma_decrease=0.1, n_gamma_decreases=100):
+        self.quantile = quantile
+        self.max_iter = max_iter
+        self.alpha = alpha
+        self.l1_ratio = l1_ratio
+        self.warm_start = warm_start
+        self.fit_intercept = fit_intercept
+        self.copy_X = copy_X
+        self.normalize = normalize
+        self.gtol = gtol
+        self.xtol = xtol
+        self.gamma = gamma
+        self.gamma_decrease = gamma_decrease
+        self.n_gamma_decreases = n_gamma_decreases
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit the model according to the given training data.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training vector, where n_samples in the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape (n_samples,)
+            Target vector relative to X.
+
+        sample_weight : array-like, shape (n_samples,)
+            Weight given to each sample.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        X, y = check_X_y(
+            X, y, copy=False, accept_sparse=['csr'], y_numeric=True)
+
+        X, y, X_offset, y_offset, X_scale = self._preprocess_data(
+            X, y, self.fit_intercept, self.normalize, self.copy_X,
+            sample_weight=sample_weight)
+
+        if sample_weight is not None:
+            sample_weight = np.array(sample_weight)
+            check_consistent_length(y, sample_weight)
+        else:
+            sample_weight = np.ones_like(y)
```

Comment:
If you merge master in you could use the `_check_sample_weights` helper form utils validation.